keras/preprocessing/image.py

```diff
@@ -337,6 +337,60 @@ def list_pictures(directory, ext='jpg|jpeg|bmp|png'):
             if re.match(r'([\w]+\.(?:' + ext + '))', f)]
 
 
+def hsv_augmentation(x, hue_shift,
+                     saturation_scale, saturation_shift,
+                     value_scale, value_shift, dim_ordering='default'):
+    """Function to perform HSV data augmentation by shifting and rescaling the HSV color representations of a batch of images
+       using values drawn from uniform distributions:
+           H += U(-hue_shift, +hue_shift)
+           S *= U(1/(1+saturation_scale), 1+saturation_scale)
+           S += U(-saturation_shift, +saturation_shift)
+           V *= U(1/(1+value_scale), 1+value_scale)
+           V += U(-value_shift, value_shift)
+    """
+    from matplotlib.colors import rgb_to_hsv, hsv_to_rgb
+
+    if dim_ordering == 'default':
+        dim_ordering = K.image_dim_ordering()
+    if dim_ordering not in {'th', 'tf'}:
+        raise ValueError('Unknown dim_ordering: ', dim_ordering)
+
+    assert x.ndim == 4, "Input of shape (examples, x, y, channels) is expected."
+    if dim_ordering == 'th':
+        x = x.transpose(0, 2, 3, 1)
+    assert x.shape[3] == 3, "RGB image with three channels is expected."
+
+    nexamples = x.shape[0]
+
+    # Assumes x has already been rescaled by (1./255)
+    hsv = rgb_to_hsv(x)
+    saturation_scale, value_scale = float(saturation_scale), float(value_scale)
+
+    hsv[..., 0] += np.random.uniform(low=-hue_shift,
```

Comment:
Array ellipsis is an obscure feature and not necessary here. It hinders readability